libvast/src/system/index.cpp

```diff
@@ -1584,17 +1591,67 @@ index(index_actor::stateful_pointer<index_state> self,
             // states and the client would go out of sync. That would require
             // the index to deal with a few complicated corner cases.
             self->state.pending_queries.mark_partition_erased(partition_id);
+            // Remove the synopsis file. We can already safely do so because
+            // the catalog acked the erase.
+            self
+              ->request<caf::message_priority::high>(self->state.filesystem,
+                                                     caf::infinite,
+                                                     atom::erase_v,
+                                                     synopsis_path)
+              .then(
+                [self, partition_id](atom::done) {
+                  VAST_DEBUG("{} erased partition synopsis {} from "
+                             "filesystem",
+                             *self, partition_id);
+                },
+                [self, partition_id, synopsis_path](const caf::error& err) {
+                  VAST_WARN("{} failed to erase partition "
+                            "synopsis {} at {}: {}",
+                            *self, partition_id, synopsis_path, err);
+                });
+            // A helper function to erase the dense index file with some
+            // logging.
+            auto erase_dense_index_file = [=] {
+              self
+                ->request<caf::message_priority::high>(
+                  self->state.filesystem, caf::infinite, atom::erase_v, path)
```

Comment:
Shouldn't this be the `store_path` that's computed below?

Reply:
Ah, nevermind

Reply:
That partition should be rotated out once we attempt to access it, as that would fail—which (I think) we handle gracefully already.